docs/tutorials/wiki2/installation.rst

```diff
@@ -215,25 +214,130 @@ On Windows
-packages. Success executing this command will show a line like the following::
+The console will show ``pip`` checking for packages and installing missing
+packages. Success executing this command will show a line like the following:
+
+.. code-block:: bash
+
+   Successfully installed Chameleon-2.24 Mako-1.0.4 MarkupSafe-0.23 \
+   Pygments-2.1.3 SQLAlchemy-1.0.12 pyramid-chameleon-0.3 \
+   pyramid-debugtoolbar-2.4.2 pyramid-mako-1.0.2 pyramid-tm-0.12.1 \
+   transaction-1.4.4 tutorial waitress-0.8.10 zope.sqlalchemy-0.7.6
+
+
+.. _install-testing-requirements:
+
+Install testing requirements
+----------------------------
+
+In order to run tests, we need to install the testing requirements, which means
+we need to edit our ``setup.py``:
+
+.. .. literalinclude:: src/installation/setup.py
+   :language: py
+   :linenos:
+   :emphasize-lines: 23-30,50-52
+
+.. code-block:: python
+   :linenos:
+   :emphasize-lines: 23-29,49-51
+
+   import os
+
+   from setuptools import setup, find_packages
+
+   here = os.path.abspath(os.path.dirname(__file__))
+   with open(os.path.join(here, 'README.txt')) as f:
+       README = f.read()
+   with open(os.path.join(here, 'CHANGES.txt')) as f:
+       CHANGES = f.read()
+
+   requires = [
+       'pyramid',
+       'pyramid_jinja2',
+       'pyramid_debugtoolbar',
+       'pyramid_tm',
+       'SQLAlchemy',
+       'transaction',
+       'zope.sqlalchemy',
+       'waitress',
+       ]
+
+   tests_require = [
+       'WebTest >= 1.3.1',  # py3 compat
+       ]
+
+   testing_extras = tests_require + [
+       'pytest',  # includes virtualenv
+       'pytest-cov',
+       ]
+
+   setup(name='tutorial',
+         version='0.0',
+         description='tutorial',
+         long_description=README + '\n\n' + CHANGES,
+         classifiers=[
+           "Programming Language :: Python",
+           "Framework :: Pyramid",
+           "Topic :: Internet :: WWW/HTTP",
+           "Topic :: Internet :: WWW/HTTP :: WSGI :: Application",
+           ],
+         author='',
+         author_email='',
+         url='',
+         keywords='web wsgi bfg pylons pyramid',
+         packages=find_packages(),
+         include_package_data=True,
+         zip_safe=False,
+         test_suite='tutorial',
+         extras_require={
+             'testing': testing_extras,
+         },
+         tests_require=tests_require,
+         install_requires=requires,
+         entry_points="""\
+         [paste.app_factory]
+         main = tutorial:main
+         [console_scripts]
+         initialize_tutorial_db = tutorial.scripts.initializedb:main
+         """,
+         )
+
+Only the emphasized lines need to be edited.
+
+Next install the testing requirements.
+
+On UNIX
+^^^^^^^
+
+.. code-block:: bash
+
+   $ $VENV/bin/pip install -e ".[testing]"
+
+On Windows
+^^^^^^^^^^
+
+.. code-block:: ps1con
+
+   c:\pyramidtut\tutorial> %VENV%\Scripts\pip install -e ".[testing]"
```

Comment:
Because this is *so* much easier and less likely to have typos than `pip install -r testing_requirements.txt`! NB this will attempt to install the package again and tell the use that it's already up to date.